README.md

```diff
@@ -2,19 +2,126 @@
 
 This server handles uploads, downloads and deletes on the storage backend for Qabel Box.
 
-Installation:
+## Requirements
+
+* Postgresql >= 9.5
+* Python >= 3.5
+* Redis
+
+## Installation
 
 	cd src
 	virtualenv ../venv --python=python3.5
 	source ../venv/bin/activate
 	pip install -r ../requirements.txt
-	python run.py
-
 
-The server needs a postgresql (>=9.5) database that needs to me initialized by
+The server needs a PostgreSQL (>=9.5) database that needs to me initialized by
 
 	alembic -x "url=postgresql://username:password@localhost/dbname" upgrade head
 
-The tests also need a redis server on localhost, running on port 6379. They start their own postgresql instance given it is installed.
-And you need to place AWS credentials in `~/.aws`, the [boto docs](https://boto3.readthedocs.io/en/latest/guide/quickstart.html#configuration)
-document details and alternatives.
+Note: If the DSN for your PostgreSQL is different, then you need to adjust it both for Alembic and set the `--psql-dsn`
+option accordingly.
+
+## Running
+
+- The tests
+
+	The tests also need a redis server on localhost, running on port 6379. They start their own postgresql instance
+	given it is installed.
+
+	And you need to place AWS credentials in `~/.aws`, the
+	[boto docs](https://boto3.readthedocs.io/en/latest/guide/quickstart.html#configuration)
+	document details and alternatives.
+
+      src$ py.test
+
+- The server directly
+
+      src$ python run.py <options>
+
+- The server in the uWSGI application server: see `src/uwsgi_plumbing.py`. Read the part about how you pass options
+  to the server.
+
+## Storage backends
+
+Available storage backends:
+
+- S3
+
+  S3 is the default and requires AWS credentials in `~/.aws`, the
+  [boto docs](https://boto3.readthedocs.io/en/latest/guide/quickstart.html#configuration) document details and
+  alternatives.
+
+- Local storage
+
+  Local storage requires nothing special, just a file system. The option is `--local-storage` (on the command line)
+  and takes the directory to store files in as a sole parameter
+
+- Dummy (temporary), `--dummy`, requires no parameters and is an amnesiac.
+
+## Options reference
+
+(from `python run.py --help`)
+
+    Tornado Logging options:
```

Comment:
This is section less important than the "Block server options" and therefore should be at the end of the output but this might be just me.